odoo/addons/openupgrade_records/models/openupgrade_record.py

```diff
@@ -119,3 +125,43 @@ def field_dump(self):
                 })
             data.append(repre)
         return data
+
+    # backward compatibility from upgrade_analysis (v14)
+    @api.model
+    def list_modules(self):
+        """ Return the set of covered modules """
+        self.env.cr.execute(
+            """SELECT DISTINCT(module) FROM openupgrade_record
+            ORDER BY module"""
+        )
+        return [module[0] for module in self.env.cr.fetchall()]
+
+    # backward compatibility from upgrade_analysis (v14)
+    @staticmethod
+    def _read_manifest(addon_dir):
+        for manifest_name in MANIFEST_NAMES:
+            if os.access(os.path.join(addon_dir, manifest_name), os.R_OK):
+                with open(os.path.join(addon_dir, manifest_name), "r") as f:
+                    manifest_string = f.read()
+                    return ast.literal_eval(manifest_string)
+        raise ValidationError(_("No manifest found in %s" % addon_dir))
+
+    # backward compatibility from upgrade_analysis (v14)
+    @api.model
+    def get_xml_records(self, module):
```

Comment:
Where do you use this?

Reply:
It's called from upgrade_analysis module when executing the analysis v13/v14.